unit_test/test_spdm_requester/get_digests.c

```diff
@@ -1577,6 +1640,91 @@ static void libspdm_test_requester_get_digests_case26(void **state)
+    spdm_context->connection_info.capability.flags |= SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CERT_CAP;
+    spdm_context->connection_info.algorithm.base_hash_algo = m_libspdm_use_hash_algo;
+    libspdm_set_mem(m_libspdm_local_certificate_chain,
+                    sizeof(m_libspdm_local_certificate_chain),
+                    (uint8_t)(0xFF));
+    libspdm_reset_message_b(spdm_context);
+
+#if LIBSPDM_RECORD_TRANSCRIPT_DATA_SUPPORT
+    spdm_context->transcript.message_m.buffer_size =
+        spdm_context->transcript.message_m.max_buffer_size;
+#endif
+    spdm_context->connection_info.multi_key_conn_rsp = true;
+    libspdm_reset_message_d(spdm_context);
+
+    m_local_key_pair_id[0] = 0x00;
+    m_local_cert_info[0] = SPDM_CERTIFICATE_INFO_CERT_MODEL_DEVICE_CERT;
+    m_local_key_usage_bit_mask[0] = SPDM_KEY_USAGE_BIT_MASK_KEY_EX_USE;
+
+    m_local_key_pair_id[1] = 0x01;
+    m_local_cert_info[1] = SPDM_CERTIFICATE_INFO_CERT_MODEL_ALIAS_CERT;
+    m_local_key_usage_bit_mask[1] = SPDM_KEY_USAGE_BIT_MASK_CHALLENGE_USE;
+
+    m_local_key_pair_id[2] = 0x02;
+    m_local_cert_info[2] = SPDM_CERTIFICATE_INFO_CERT_MODEL_GENERIC_CERT;
+    m_local_key_usage_bit_mask[2] = SPDM_KEY_USAGE_BIT_MASK_MEASUREMENT_USE;
+
+    m_local_key_pair_id[3] = 0x03;
+    m_local_cert_info[3] = SPDM_CERTIFICATE_INFO_CERT_MODEL_DEVICE_CERT;
+    m_local_key_usage_bit_mask[3] = SPDM_KEY_USAGE_BIT_MASK_ENDPOINT_INFO_USE;
+
+    m_local_key_pair_id[4] = 0x04;
+    m_local_cert_info[4] = SPDM_CERTIFICATE_INFO_CERT_MODEL_NONE;
+    m_local_key_usage_bit_mask[4] = SPDM_KEY_USAGE_BIT_MASK_STANDARDS_KEY_USE;
+
+    m_local_key_pair_id[5] = 0x05;
+    m_local_cert_info[5] = SPDM_CERTIFICATE_INFO_CERT_MODEL_GENERIC_CERT;
+    m_local_key_usage_bit_mask[5] = SPDM_KEY_USAGE_BIT_MASK_VENDOR_KEY_USE;
+
+    slot_count = SPDM_MAX_SLOT_COUNT;
+    additional_size = sizeof(spdm_key_pair_id_t) + sizeof(spdm_certificate_info_t) +
+                      sizeof(spdm_key_usage_bit_mask_t);
+    hash_size = libspdm_get_hash_size(m_libspdm_use_hash_algo);
+
+    libspdm_zero_mem(total_digest_buffer, sizeof(total_digest_buffer));
+    status = libspdm_get_digest(spdm_context, NULL, &slot_mask, &total_digest_buffer);
+    assert_int_equal(status, LIBSPDM_STATUS_SUCCESS);
+    assert_int_equal(
+        spdm_context->transcript.message_d.buffer_size,
+        sizeof(spdm_digest_response_t) + (additional_size + hash_size) * slot_count);
+
+    for (uint8_t index = 0; index < SPDM_MAX_SLOT_COUNT; index++) {
+        assert_memory_equal((void *)&m_local_key_pair_id[index],
+                            (void *)&spdm_context->connection_info.peer_key_pair_id[index],
+                            sizeof(spdm_key_pair_id_t));
+        assert_memory_equal((void *)&m_local_cert_info[index],
+                            (void *)&spdm_context->connection_info.peer_cert_info[index],
+                            sizeof(spdm_key_pair_id_t));
```

Comment:
Line 1721 `sizeof(spdm_key_pair_id_t)` should be `sizeof(spdm_certificate_info_t)`.